qiskit/circuit/qpy_serialization.py

```diff
@@ -316,6 +361,8 @@
 you then read the`metadata_size`` number of bytes and parse the JSON to get
 the metadata for the circuit.
 
+.. _registers:
+
```

Comment:
Are Sphinx's scoping rules for references safe for simple things like this?  I don't really remember what they are, but I remember worrying about it a bit when we started putting in generic reference names like `version 1`.  Sorry I don't have time to check myself immediately - I'll do it on Monday.

Reply:
Oh good call, I'm not sure either. I know they're exposed globally, since you use it to do cross-document references, but I don't know off the top of my head how it handles scoping it might just error if tehre are two refs with the same. I can rename this to `qpy_version_3` to make it more descriptive though.

Reply:
Done in: https://github.com/Qiskit/qiskit-terra/pull/7584/commits/d378f7793163f7cb1e213c74c155d33bed7aa997